Adaptors/Nats/src/QueueMessageHandler.cs

```diff
@@ -0,0 +1,134 @@
+// This file is part of the ArmoniK project
+// 
+// Copyright (C) ANEO, 2021-2025. All rights reserved.
+// 
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+// 
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY, without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+// 
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+using ArmoniK.Core.Base;
+
+using Microsoft.Extensions.Logging;
+
+using NATS.Client.JetStream;
+
+namespace ArmoniK.Core.Adapters.Nats;
+
+internal class QueueMessageHandler : IQueueMessageHandler
+{
+  private readonly Heart             autoExtendAckDeadline_;
+  private readonly INatsJSContext    js_;
+  private readonly ILogger           logger_;
+  private readonly NatsJSMsg<string> message_;
+  private          StackTrace?       stackTrace_;
+
+
+  public QueueMessageHandler(NatsJSMsg<string> message,
+                             INatsJSContext    js,
+                             int               AckWait,
+                             int               ackExtendDeadlineStep,
+                             ILogger           logger,
+                             CancellationToken cancellationToken)
+  {
+    js_               = js;
+    message_          = message;
+    MessageId         = message.Data!;
+    TaskId            = message.Data!;
```

Comment:
it should not be the same. It could be a GUID given at the push of the message if the queue system supports it (or it already exists)